staging/src/k8s.io/client-go/kubernetes/clientset.go

```diff
@@ -436,6 +439,17 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
 	return c.DiscoveryClient
 }
 
+// WithLegacyDiscovery retrieves a DiscoveryClient which will only
+// receive the legacy discovery file format.
+func (c *Clientset) WithLegacyDiscovery() *discovery.DiscoveryClient {
+	if c == nil {
+		return nil
+	}
+	// Forces this client to request only "unaggregated" (legacy) discovery.
+	c.DiscoveryClient.UseLegacyDiscovery = true
+	return c.DiscoveryClient
+}
```

Comment:
I'm very tedious about the function call that alters the internal discovery client. That sequence of events sounds wrong:
```
discovery := clientset.Discovery() // that's aggregated discovery
discovery = clientset.WithLegacyDiscovery() // that's unaggregated discovery
discovery = clientset.Discovery() // that's unaggregated discovery!?

```

Reply:
To be honest, the flag is not threadsafe either I suspect, so I don't know if we should ever change it after the client has been shared across different actors. Can we have two Discovery?

Reply:
Sorry for not being clear, I trying to ask for `clientset.Discovery().WithLegacy()`

Reply:
And it should make a copy, or return a wrapper object with the setting, not change the base object.